datastore/migrations/core/migrater_memory.py

```diff
@@ -0,0 +1,141 @@
+from datetime import datetime
+from typing import Dict, List, Tuple, cast
+
+from datastore.shared.di import service_as_factory
+from datastore.shared.postgresql_backend import ConnectionHandler
+from datastore.shared.services import ReadDatabase
+from datastore.shared.typing import Fqid, Model, Position
+
+from .base_migration import BaseMigration
+from .events import BaseEvent, CreateEvent
+from .exceptions import MismatchingMigrationIndicesException
+from .migrater import RawPosition
+from .migration_keyframes import (
+    InitialMigrationKeyframeModifier,
+    MigrationKeyframeModifier,
+)
+from .migration_logger import MigrationLogger
+
+
+@service_as_factory
+class MigraterImplementationMemory:
+    read_database: ReadDatabase
+    connection: ConnectionHandler
+    logger: MigrationLogger
+    target_migration_index: int
```

Comment:
add `start_migration_index` declaration here for consistency

Reply:
Done